Vi's New Alcohol

Vimas21 opened the pull request:
:cl: SammasC123

rscadd: New mixed drinks, including three alcohols to the booze dispenser: Crème de Menthe, Crème de Cacao, and Triple Sec. 

/:cl:

SierraKomodo reviewed (changes requested):
Changelog format isn't valid - The change needs a prefix. See https://tgstation13.org/wiki/Guide_to_Changelogs for examples.

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:
```diff
@@ -1340,6 +1340,26 @@
+/datum/reagent/ethanol/creme_de_menthe
+	name = "Creme de Menthe"
+	description = "A sweet, mint-flavored alcoholic beverage"
+	taste_description = "a melted breathmint"
```
Avoid usage of `a`/`an` in taste descriptions - `melted breathmints` would read better in the generated text.
```suggestion
	taste_description = "melted breathmints"
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:
```diff
@@ -1460,6 +1480,16 @@
+	name = "Triple Sec"
+	description = "It smells like oranges."
+	taste_description = "sweet citrus"
+	color = "#fac840" // rgb: 145, 255, 145
```
RGB comment isn't really necessary here

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:
```diff
@@ -1558,6 +1588,16 @@
 
+/datum/reagent/ethanol/alexander
+	name = "Alexander"
+	description = "Better be careful- it's stronger than it looks..."
```
Missing a space before the `-`

```suggestion
	description = "Better be careful - it's stronger than it looks..."
```

Vimas21 commented:
> Changelog format isn't valid - The change needs a prefix. See https://tgstation13.org/wiki/Guide_to_Changelogs for examples.

Fixed.

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:
```diff
@@ -1661,6 +1711,21 @@
+
+/datum/reagent/ethanol/between_the_sheets/affect_ingest(mob/living/carbon/M, alien, removed)
+	. = ..()
+	if(prob(1)) // Rarely makes the user fall asleep... between the sheets. ;)
```
Consider checks for certain species that should probably not be affected by or be more sensitive to alcohol, such as Diona and Skrell. I suggest talking to the maintainers for their take on this - @EcklesFire for Skrell and @purplemontart for Diona.

See the parent proc in `/datum/reagent/ethanol/affect_ingest()` for examples of how these two species affect existing alcohol procs.

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:
```diff
@@ -1743,6 +1808,16 @@
 
+/datum/reagent/ethanol/cobalt_velvet
+	name = "Cobalt Velvet"
+	description = "It's like champaigne served in a cup that had a bit of cola left."
```
Mispelled
```suggestion
	description = "It's like champagne served in a cup that had a bit of cola left."
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:
```diff
@@ -1743,6 +1808,16 @@
+	strength = 40
+
+	glass_name = "Cobalt Velvet"
+	glass_desc = "It's like champaigne served in a cup that had a bit of cola left."
```
Mispelled
```suggestion
	glass_desc = "It's like champagne served in a cup that had a bit of cola left."
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:
```diff
@@ -1775,6 +1850,36 @@
 	glass_name = "Driest Martini"
 	glass_desc = "Only for the experienced. You think you see sand floating in the glass."
 
+/datum/reagent/ethanol/forget_me_shot
+	name = "Forget Me Shot"
+	description = "If you haven't heard of it, that means it worked."
+	taste_description = "mind-numbing venom"
+	color = "#a8934b"
+	strength = 5
+	var/already_used = FALSE
+
+	glass_name = "Forget Me Shot"
+	glass_desc = "Mixed with a pregnancy test until it turned positive."
+
+/datum/reagent/ethanol/forget_me_shot/affect_ingest(mob/living/carbon/M, alien, removed)
+	. = ..()
+	if(alien == IS_DIONA)
+		return
+	if(istype(M, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = M
+		if(H.has_brain())
+			M.confused = max(M.confused, 10)		
+			var/obj/item/organ/internal/brain/brain = H.internal_organs_by_name[BP_BRAIN]
+			M.make_dizzy(3)
+			M.sleeping = max(M.sleeping, 3)
+			M.eye_blurry = max(M.eye_blurry, 10)
+			if(already_used || SSkv.Get("forget_me_shot", M))
+				return
+			//one-shot things here- shoutout to spookerton for getting the code working, love ya.
+			SSkv.Put("forget_me_shot", TRUE, M)
+			already_used = TRUE
+			brain.handle_severe_brain_damage()
+
```
I'm not sure how I feel about a drink that procs severe brain damage like this - It could have unintended effects. I'll have to dig into and think about this.

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:
```diff
@@ -1785,6 +1890,16 @@
 
+/datum/reagent/ethanol/grasshopper
+	name = "Grasshopper"
+	description = "Like a chocolate mint, but alcoholic!."
```
Unnecessary period
```suggestion
	description = "Like a chocolate mint, but alcoholic!"
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:
```diff
@@ -1785,6 +1890,16 @@
+/datum/reagent/ethanol/grasshopper
+	name = "Grasshopper"
+	description = "Like a chocolate mint, but alcoholic!."
+	taste_description = "minty chocolates"
```
```suggestion
	taste_description = "minty chocolate"
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:
```diff
@@ -2072,6 +2217,26 @@
 			else
 				L.take_internal_damage(100, 0)
 
+/datum/reagent/ethanol/quadruple_sec
+	name = "Quadruple Sec"
+	description = "Red, like lethals."
+	taste_description = "protecting the station"
+	color = "#ac1515"
+	strength = 20
+
+	glass_name = "Quadruple Sec"
+	glass_desc = "Protect the weak and the innocent. Hurt everyone else."
+
+/datum/reagent/ethanol/quintuple_sec
+	name = "Quintuple Sec"
+	description = "Red, like blood on securities boots."
+	taste_description = "violence"
+	color = "#fa0b0b"
+	strength = 5
+
+	glass_name = "Quintuple Sec"
+	glass_desc = "I am the last bastion between chaos and order."
+
```
These feel a bit too memey for an HRP server's codebase.

Spookerton replied:
Yeah, that's a no on this and the above.

SierraKomodo commented on code/modules/reagents/dispenser/dispenser_presets.dm:
```diff
@@ -117,7 +117,10 @@
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/mead,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/triple_sec,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/creme_de_cacao,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/creme_de_menthe
```
Add a trailing comma at the end - It helps keep the git history clean if future additions are made.
```suggestion
			/obj/item/weapon/reagent_containers/chem_disp_cartridge/creme_de_menthe,
```

Spookerton commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:
```diff
@@ -1775,6 +1850,36 @@
+			M.make_dizzy(3)
+			M.sleeping = max(M.sleeping, 3)
+			M.eye_blurry = max(M.eye_blurry, 10)
+			if(already_used || SSkv.Get("forget_me_shot", M))
+				return
+			//one-shot things here- shoutout to spookerton for getting the code working, love ya.
+			SSkv.Put("forget_me_shot", TRUE, M)
+			already_used = TRUE
```
will have problems with signature changes in #29415 + doesn't incorporate the dumb thing I forgot in the example. This will be fine instead.

```suggestion
			if (already_used)
				return
			already_used = SSkv.Get(M, "forget_me_shot")
			if (already_used)
				return
			SSkv.Put(M, "forget_me_shot", TRUE)
			already_used = TRUE
```

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:
```diff
@@ -1661,6 +1711,26 @@
+	if(alien == IS_SKRELL)
+		strength_mod *= 2
+	if(alien == IS_DIONA)
+		strength_mod = 0
```
Better to just `return` here - `prob(1*0)` is either never going to pass or is going to do something weird, knowing byond.

SierraKomodo commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:
```diff
@@ -1775,6 +1855,37 @@
 	glass_name = "Driest Martini"
 	glass_desc = "Only for the experienced. You think you see sand floating in the glass."
 
+/datum/reagent/ethanol/forget_me_shot
+	name = "Forget Me Shot"
+	description = "If you haven't heard of it, that means it worked."
+	taste_description = "mind-numbing venom"
+	color = "#a8934b"
+	strength = 5
+	var/already_used = FALSE
+
+	glass_name = "Forget Me Shot"
+	glass_desc = "Mixed with a pregnancy test until it turned positive."
+
+/datum/reagent/ethanol/forget_me_shot/affect_ingest(mob/living/carbon/M, alien, removed)
+	. = ..()
+	if(alien == IS_DIONA)
+		return
+	if(istype(M, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = M
+		if(H.has_brain())
+			M.confused = max(M.confused, 10)		
+			var/obj/item/organ/internal/brain/brain = H.internal_organs_by_name[BP_BRAIN]
+			M.make_dizzy(3)
+			M.sleeping = max(M.sleeping, 3)
+			M.eye_blurry = max(M.eye_blurry, 10)
+			if (already_used)
+				return
+			already_used = SSkv.Get(M, "forget_me_shot")
+			if (already_used)
+				return
+			SSkv.Put(M, "forget_me_shot", TRUE)
+			already_used = TRUE
+
```
Still unsure about this one. Checking with other devs/admins and will get back to you.

SierraKomodo replied:
Okay this one's going to be a hard no - Remove the 'forget me shot'.

Vimas21 commented:
All of forget-me-shots effects are removed, drink recipe changed (removed mindbreaker toxin, added hooch). Now it's just a strong alcohol for the enterprising bartender.

Spookerton commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:
```diff
@@ -1661,6 +1711,26 @@
+		return
+	if(alien == IS_SKRELL)
+		strength_mod *= 2
+	if(prob(1*strength_mod)) // Rarely makes the user fall asleep... between the sheets. ;)
```
```suggestion
	if(prob(1*strength_mod))
```

Spookerton commented on code/modules/reagents/dispenser/dispenser_presets.dm:
```diff
@@ -117,7 +117,10 @@
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/mead,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/triple_sec,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/creme_de_cacao,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/creme_de_menthe,
```
```suggestion
			/obj/item/weapon/reagent_containers/chem_disp_cartridge/creme_de_menthe
```

Spookerton commented on code/modules/reagents/Chemistry-Recipes.dm:
```diff
@@ -2629,6 +2629,75 @@
+/datum/chemical_reaction/forget_me_shot
+	name = "Forget-me-shot"
+	result = /datum/reagent/ethanol/forget_me_shot
+	required_reagents = list(/datum/reagent/ethanol/jagermeister = 1, /datum/reagent/ethanol/gin = 1, /datum/reagent/ethanol/triple_sec = 1, /datum/reagent/ethanol/quadruple_sec= 1, /datum/reagent/nutriment/vinegar=1, /datum/reagent/ethanol/hooch = 1)
```
```suggestion
	required_reagents = list(/datum/reagent/ethanol/jagermeister = 1, /datum/reagent/ethanol/gin = 1, /datum/reagent/ethanol/triple_sec = 1, /datum/reagent/nutriment/vinegar =1, /datum/reagent/ethanol/hooch = 1)
```

Spookerton commented on code/modules/reagents/Chemistry-Recipes.dm:
```diff
@@ -2629,6 +2629,75 @@
+/datum/chemical_reaction/between_the_sheets
+	name = "Between the Sheets"
+	result = /datum/reagent/ethanol/between_the_sheets
+	required_reagents = list(/datum/reagent/ethanol/rum = 1, /datum/reagent/ethanol/cognac=2, /datum/reagent/ethanol/triple_sec=1, /datum/reagent/drink/juice/lemon = 1)
```
```suggestion
	required_reagents = list(/datum/reagent/ethanol/rum = 1, /datum/reagent/ethanol/cognac = 2, /datum/reagent/ethanol/triple_sec = 1, /datum/reagent/drink/juice/lemon = 1)
```

Spookerton commented on code/modules/reagents/Chemistry-Recipes.dm:
```diff
@@ -2629,6 +2629,75 @@
+	required_reagents = list(/datum/reagent/ethanol/creme_de_menthe = 1, /datum/reagent/ethanol/cognac = 1, /datum/reagent/drink/milk/cream = 1)
+	result_amount = 3
+
+/datum/chemical_reaction/quadruple_sec
+	name = "Quadruple Sec"
+	result = /datum/reagent/ethanol/quadruple_sec
+	required_reagents = list(/datum/reagent/ethanol/triple_sec = 1, /datum/reagent/drink/triplecitrus = 1, /datum/reagent/ethanol/creme_de_menthe = 1)
+	result_amount = 3
+
+/datum/chemical_reaction/quintuple_sec
+	name = "Quintuple Sec"
+	result = /datum/reagent/ethanol/quintuple_sec
+	required_reagents = list(/datum/reagent/ethanol/quadruple_sec = 5, /datum/reagent/ethanol/screwdrivercocktail = 5, /datum/reagent/drink/juice/orange = 5)
+	result_amount = 5 // Because Quintuple sec is incredibly concentrated.
+	mix_message = "The drink sizzles, and a small puff of orange smoke emerges."
+
```
```suggestion
```

Spookerton commented on code/modules/reagents/Chemistry-Recipes.dm:
```diff
@@ -2629,6 +2629,75 @@
 	required_reagents = list(/datum/reagent/drink/milk/cream = 1, /datum/reagent/ethanol/cognac = 1, /datum/reagent/drink/hot_coco = 1)
 	result_amount = 3
 
+// Alcohol Expansion, Electric Boogaloo
+
```
```suggestion
```

Spookerton commented on code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm:
```diff
@@ -2072,6 +2202,26 @@
 			else
 				L.take_internal_damage(100, 0)
 
+/datum/reagent/ethanol/quadruple_sec
+	name = "Quadruple Sec"
+	description = "The smell of oranges is overwhelming."
+	taste_description = "fermented oranges"
+	color = "#d47800"
+	strength = 20
+
+	glass_name = "Quadruple Sec"
+	glass_desc = "Orange juice, with extra orange."
+
+/datum/reagent/ethanol/quintuple_sec
+	name = "Quintuple Sec"
+	description = "A metric ton of oranges in every drop."
+	taste_description = "overwhelming, absolute orange"
+	color = "#fa430b"
+	strength = 5
+
+	glass_name = "Quintuple Sec"
+	glass_desc = "There is more orange in this drink than what should be physically possible."
+
```
```suggestion
```

SierraKomodo reviewed (commented):
Still a couple of unresolved items from `dispenser_presets.dm` and `Chemistry-Reagents-Food-Drinks.dm`

SierraKomodo commented:
Just noticed me and Spookerton are giving conflicting requests. Gonna work that one out with him, sorry.

SierraKomodo commented:
Alright decision is to roll with Spook's version of changes.

Vimas21 commented:
> Alright decision is to roll with Spook's version of changes.

What changes are those? You both made lots, and I don't think they conflicted?

Vimas21 committed:
So much new alcohol

so much
More Cool shit!


Update


Forget me shot, now just a normal drink

Recipe changed, as well!
Apply suggestions from code review

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/modules/reagents/dispenser/dispenser_presets.dm

Co-authored-by: Sierra Brown (SierraKomodo) <SierraKomodo@users.noreply.github.com>
Update code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
There's No Comma now

there's no comma now

Vimas21 commented:
Ready for Review

SierraKomodo reviewed (commented):
I'm hesitant about adding more drinks to an already bloated file, but from a code standpoint, this looks good now. If another dev approves I have no issues merging.

SierraKomodo reviewed (approved):
You know what, let's do it.
